beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -486,19 +530,68 @@ impl<T: BeaconChainTypes> DataAvailabilityCheckerInner<T> {
             .unwrap_or_else(|| PendingComponents::empty(block_root));
 
         // Merge in the data columns.
-        pending_components.merge_data_columns(kzg_verified_data_columns);
+        pending_components.merge_data_columns(kzg_verified_data_columns)?;
 
         let block_import_requirement = self.block_import_requirement(epoch)?;
+
+        // Potentially trigger reconstruction if:
+        // - Our custody requirement is all columns
+        // - We >= 50% of columns
+        let data_columns_to_publish =
+            if self.should_reconstruct(&block_import_requirement, &pending_components) {
+                pending_components.reconstruction_started();
+
+                let timer = metrics::start_timer(&metrics::DATA_AVAILABILITY_RECONSTRUCTION_TIME);
+
+                let existing_column_indices = pending_components
+                    .verified_data_columns
+                    .iter()
+                    .map(|d| d.index())
+                    .collect::<HashSet<_>>();
+
+                // Will only return an error if:
+                // - < 50% of columns
+                // - There are duplicates
+                let all_data_columns = KzgVerifiedCustodyDataColumn::reconstruct_columns(
```

Comment:
Yeah good call - I've got a PR that refactors the reconstruction code, and reconstruction has been lifted out of DA checker, it might be easier to include the change there I think
https://github.com/sigp/lighthouse/pull/5990

Reply:
Sounds good 👍